pkg/storage/ibmcos/ibmcos.go

```diff
@@ -0,0 +1,659 @@
+package ibmcos
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/aws/awserr"
+	"github.com/IBM/ibm-cos-sdk-go/aws/credentials/ibmiam"
+	"github.com/IBM/ibm-cos-sdk-go/aws/session"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3/s3manager"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
+	configapiv1 "github.com/openshift/api/config/v1"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/envvar"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+)
+
+const (
+	IAMEndpoint                   = "https://iam.cloud.ibm.com/identity/token"
```

Comment:
I do wonder if hard-coding some of these endpoints will come back to haunt us in the future.
If hard-coding is a must, is there some way to define the endpoint in fewer places or is there a way to override it?
Seems like this precludes us from running this in staging against the `https://iam.test.cloud...` instance.